tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -1648,6 +1665,19 @@ impl Display for TimingDisplay {
     }
 }
 
+struct FormatTimeDisplay<T>(T);
+impl<T: FormatTime> fmt::Display for FormatTimeDisplay<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // If getting the timestamp failed, don't bail --- only bail on
+        // formatting errors.
+        if self.0.format_time(&mut Writer::new(f)).is_err() {
+            f.write_str("<unknown time>")
+        } else {
+            Ok(())
+        }
+    }
+}
+
```

Comment:
This ended up being the relatively nice workaround for https://github.com/jam1garner/owo-colors/issues/123